completion.go

```diff
@@ -0,0 +1,145 @@
+package kong
+
+import (
+	"reflect"
+)
+
+//CompletionOptions options for shell completion.
+type CompletionOptions struct {
+	// the function to run for shell completions. If nil, no completions will be run. default: nil
+	Completer Completer
+
+	// installer/uninstaller for shell completions.  If nil, there will be no install or uninstall flags
+	CompletionInstaller CompletionInstaller
+
+	// flag name for installing completions to a shell.  default: --install-shell-completion
+	InstallFlagName string
+
+	// help text for --install-shell-completion. default: "Install shell completion."
+	InstallFlagHelp string
+
+	// flag name for uninstalling completions from a shell.  default: --uninstall-shell-completion
+	UninstallFlagName string
+
+	// help text for --uninstall-shell-completion. default: "Uninstall shell completion."
+	UninstallFlagHelp string
+
+	// whether to hide --install-shell-completion and --uninstall-shell-completion from help.  default: false
+	HideFlags bool
+}
+
+func (c CompletionOptions) withDefaults() CompletionOptions {
+	if c.InstallFlagName == "" {
+		c.InstallFlagName = "install-shell-completion"
+	}
+	if c.UninstallFlagName == "" {
+		c.UninstallFlagName = "uninstall-shell-completion"
+	}
+	if c.InstallFlagHelp == "" {
+		c.InstallFlagHelp = "Install shell completion."
+	}
+	if c.UninstallFlagHelp == "" {
+		c.UninstallFlagHelp = "Uninstall shell completion."
+	}
+	return c
+}
+
+// Completer is a function to run shell completions. Returns true if this was a completion run. Kong will exit 0
+// immediately when it returns true.
+type Completer func(*Context) (bool, error)
```

Comment:
Shouldn't this accept positional information of some sort? Or I guess that is part of the remaining integration in `kongcomplete`?

Reply:
That is handled by [PositionalPredictor](https://github.com/WillAbides/kongplete/blob/19e042f32715fd238070f8f8e3c23ad6024ed8ef/internal/positionalpredictor/positional.go) in kongplete.

Reply:
This type needs documentation on how it conveys completion information. It looks, from the tests, like it does this via stdout, but the comments need to be explicit about how and what is expected.